src/API/src/export/export.service.ts

```diff
@@ -0,0 +1,32 @@
+import { Inject, Injectable, Logger } from '@nestjs/common';
+import { OccurrenceService } from 'src/db/occurrence/occurrence.service';
+import { flattenOccurrenceRepoObject } from './utils/allDataCsvCreation';
+import { occurrenceMapper } from './utils/occurrenceMapper';
+const fs = require('fs');
+
+@Injectable()
+export class ExportService {
+  constructor(
+    @Inject(OccurrenceService)
+    private readonly occurrenceService: OccurrenceService,
+    private logger: Logger,
+  ) {}
+
+  async exportOccurrenceDbtoCsvFormat() {
+    const dbData = await this.occurrenceService.findAll();
+    try {
+      const repoCsv = await flattenOccurrenceRepoObject(dbData);
+      return repoCsv;
+    } catch (e) {
+      this.logger.error(e);
+      throw e;
+    }
+  }
+  exportCsvToDownloadsFile(csvObject: object, repo: string) {
+    const csvString = occurrenceMapper(csvObject);
+    fs.writeFileSync(
```

Comment:
Do we want this synchronous? If it fails for whatever reason I don't think we'd be notified. Though I guess this is a separate process anyway so maybe that doesn't matter?

Reply:
Can we change it to `writeFile` to make it async?